### ISSUE TITLE
[Bungee] Use LoginEvent instead of PreLoginEvent

### ISSUE THREAD

**Issue by games647** [Th3Tr0LLeR/SkinsRestorer---Maro#132]
The PreLoginEvent is called even before a premium authentication happens. This means that BungeeCord will override skin property if online mode is activated. You should rather use the LoginEvent. It's an AsyncEvent too, but it's called after a premium authentication and could possible already contain a premium skin.

Summary:
- Both events are called on login for online and offline mode servers
- LoginEvent could possible the premium skin for mixed-offline mode servers (-> reduce the number of api requests)
- Both events are async and therefore no only the method parameter should be changed

**Comment by Th3Tr0LLeR** [comment]
Yeah. Once i was checking the events which are called on join, but i didn't notice that.

Thanks

**Comment by Blackfire62** [comment]
And yes, that was why I was using it previously, but it doesnt really matter which event you use, since most plugins which need SkinsRestorer do not have online mode true in config (this one is checked after preloginevent) and those servers who need it either use online mode false (doesnt really matter which event you use) or a plugin which enables online mode per player (since online mode can still be set even after loginevent, its best to leave our listener into low login event, in case of these plugins are badly coded and use login event for online mode changing.

**Comment by games647** [comment]
@Blackfire62 
>plugins which need SkinsRestorer do not have online mode true in config

A lot of servers have mixed online mode with AutoIn or FastLogin.

>since online mode can still be set even after loginevent

But then no premium authentication happens

What's really the difficult by using the LoginEvent instead of PreLoginEvent? It's just renaming the method parameter.

**Comment by Blackfire62** [comment]
I agreed youre right. If you check my fork I should still use LoginEvent there. But it doesnt really matter even after all.

Since, when you use online mode on bungeecord, minecraft server has the final word on which skin will the player have. Bungees loginProfile is not taken into account. Thats why, when you want to change your skin while in online mode on bungeecord, you have to let the minecraft server know about it so it can set the gameprofile propeties (which matter here) . 

And skin gets auto updated after login, you ll blink and have latest skin (or from db if limited) .

**Comment by games647** [comment]
>Since, when you use online mode on bungeecord, minecraft server has the final word on which skin will the player have. Bungees loginProfile is not taken into account. Thats why, when you want to change your skin while in online mode on bungeecord, you have to let the minecraft server know about it so it can set the gameprofile propeties (which matter here) .

You can override the login profile on BungeeCord only, but it only works on login. It really works. I tried it out. You just have to have ipForward and bungeecord support (in spigot) enabled.

**Comment by Blackfire62** [comment]
Dunno, when I was testing, the skin was still overridden in online mode connection but worked fine with offline mode connection. Cant test, i dont have a computer rn.

Either way, it works.
